File: app/src/main/java/com/returnnotfound/stain/data/repository/AnimeRepository.java
package com.returnnotfound.stain.data.repository;

import com.returnnotfound.stain.data.model.AnimeTop;

import java.util.List;

import io.reactivex.Single;

public interface AnimeRepository {
  Single<List<AnimeTop>> getAnimeTop(int page, String subType);
}

File: app/src/main/java/com/returnnotfound/stain/data/repository/AnimeRepository.kt
package com.returnnotfound.stain.data.repository

import com.returnnotfound.stain.data.model.AnimeTop
import io.reactivex.Single

interface AnimeRepository {
  fun getAnimeTop(
    page: Int,
    subType: String?
  ): Single<List<AnimeTop?>?>?
}